portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java

```diff
@@ -53,21 +57,40 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String requestData = request.getReader().lines().collect(Collectors.joining(System.lineSeparator()));
-    String commentString = requestData.replaceAll("(\r\n|\n)", "<br />");
-    Entity commentEntity = Comment.createCommentEntity(commentString);
+    String[] requestData = request.getReader().lines().collect(Collectors.joining(System.lineSeparator())).split("\\n+");
+    String[] commentData = getCommentData(requestData);
+    String userId = request.getParameter("user");
+    String username = commentData[0];
+    String content = commentData[1];
+    Entity commentEntity = Comment.createCommentEntity(userId, username, content);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(commentEntity);
     doGet(request, response);
   }
 
   @Override
   public void doDelete(HttpServletRequest request, HttpServletResponse response) throws IOException {
```

Comment:
Future task:  As discussed earlier, make a status column, set status=DELETED, and only retrieve status={whatever you call not deleted} entities.

You may be tempted to do this as a boolean but status is a very common thing to add more of later, so an enum will save you trouble down the road.  For instance if you want to make comments private or frozen or something.